src/whenVisible.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * Use this function to delay execution of something until an element is visible
+ * in the viewport
+ *
+ * @param marker : The html element that we want to observe;
+ * @param callback : This is fired when the marker is visible in the viewport
+ */
+export const whenVisible = (
+	marker: HTMLElement,
+	callback: () => void,
+): void => {
+	const io = new IntersectionObserver(([entry]) => {
+		if (!entry.isIntersecting) return;
+		// Disconnect this IntersectionObserver once seen
+		io.disconnect();
+		callback();
+	});
+
+	io.observe(marker);
+};
```

Comment:
could rename `marker` to `element`? it seems a more natural way of expressing it (e.g. it's used in the description) and it will pop up in intellisense etc

```suggestion
/**
 * Use this function to delay execution of something until an element is visible
 * in the viewport
 *
 * @param element : The html element that we want to observe;
 * @param callback : This is fired when the marker is visible in the viewport
 */
export const whenVisible = (
	element: HTMLElement,
	callback: () => void,
): void => {
	const io = new IntersectionObserver(([entry]) => {
		if (!entry.isIntersecting) return;
		// Disconnect this IntersectionObserver once seen
		io.disconnect();
		callback();
	});

	io.observe(element);
};
```

Reply:
You're right, `element` is more semantic.

For some reason I never considered using suggested change over multiple lines... But this is perfect, thanks for this!